runelite-client/src/main/java/net/runelite/client/plugins/crystalmathlabs/CrystalMathLabs.java

```diff
@@ -0,0 +1,114 @@
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.crystalmathlabs;
+
+import com.google.common.eventbus.Subscribe;
+import lombok.extern.slf4j.Slf4j;
+import javax.inject.Inject;
+import net.runelite.http.api.RuneLiteAPI;
+import okhttp3.Response;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import net.runelite.api.Client;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.api.GameState;
+import net.runelite.api.events.GameStateChanged;
+import net.runelite.api.events.GameTick;
+import net.runelite.client.plugins.Plugin;
+
+@PluginDescriptor(name = "Crystal Math Labs", enabledByDefault = false)
+@Slf4j
+public class CrystalMathLabs extends Plugin
+{
+	@Inject
+	private Client client;
+
+	private String loggedInUsername = null;
+	private boolean shouldFire = false;
+	private boolean isOnLoginScreen = false;
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		GameState state = client.getGameState();
+
+		if (state == GameState.LOGGED_IN)
+		{
+			loggedInUsername = client.getLocalPlayer().getName();
+		}
+	}
+
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged gameStateChanged)
+	{
+		GameState state = gameStateChanged.getGameState();
+		if (state == GameState.LOGIN_SCREEN)
+		{
+			isOnLoginScreen = true;
+			makeApiCall();
+		}
+		else if (state == GameState.LOGGED_IN && isOnLoginScreen)
+		{
+			isOnLoginScreen = false;
+			shouldFire = true;
+		}
+	}
+
+	@Subscribe
+	public void onTick(GameTick tick)
+	{
+		if (shouldFire)
+		{
+			shouldFire = false;
+			loggedInUsername = client.getLocalPlayer().getName();
+			makeApiCall();
+		}
+	}
+
+	private void makeApiCall()
+	{
+		if (loggedInUsername == null)
+		{
+			return;
+		}
+
+		String reformedUsername = loggedInUsername.replace(" ", "_");
+		OkHttpClient httpClient = RuneLiteAPI.CLIENT;
+
+		Request request = new Request.Builder()
+				.url("https://crystalmathlabs.com/tracker/api.php?type=update&player=" + reformedUsername)
+				.build();
+		try
+		{
+			final Response resp = httpClient.newCall(request).execute();
```

Comment:
The request should be done the okhttp worker thread.

Reply:
Where can i find this okhttp worker thread?

Reply:
Should be fixed.